exporter/loadbalancingexporter/loadbalancer.go

```diff
@@ -153,8 +152,9 @@ func endpointWithPort(endpoint string) string {
 func (lb *loadBalancerImp) removeExtraExporters(ctx context.Context, endpoints []string) {
 	for existing := range lb.exporters {
 		if !endpointFound(existing, endpoints) {
-			lb.exporters[existing].Shutdown(ctx)
-			delete(lb.exporters, existing)
+			if err := lb.exporters[existing].Shutdown(ctx); err != nil {
```

Comment:
I think the flow is not right here. It should be the other way around at least:

```suggestion
			if err := lb.exporters[existing].Shutdown(ctx); err == nil {
```

But preferably, to not change the flow of the code, we should just log the failure if it happens and delete the exporter nevertheless